test/src/test_thread.rs

```diff
@@ -11,7 +11,7 @@ pub fn threaded_fac<'a>(env: NifEnv<'a>, args: &Vec<NifTerm>) -> NifResult<NifTe
     }
 
     let n: u64 = args[0].decode()?;
-    thread::spawn(env, move |thread_env| {
+    thread::spawn::<thread::ThreadSpawner, _>(env, move |thread_env| {
```

Comment:
I don't see a way to avoid it while keeping `spawn` as its own function. It might make sense to offer it only as a method of the trait, though.